internal/topology/variables/clusterclass_variable_validation_test.go

```diff
@@ -1023,12 +1023,266 @@ func Test_ValidateClusterClassVariable(t *testing.T) {
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "object",
+						XValidations: []clusterv1.ValidationRule{
+							{
+								Rule:      "self.a > 0",
+								Reason:    clusterv1.FieldValueErrorReason("InternalError"),
+								FieldPath: ".a",
+							},
+						},
+						Properties: map[string]clusterv1.JSONSchemaProps{
+							"a": {
+								Type: "number",
+								XValidations: []clusterv1.ValidationRule{
+									{
+										Rule:   "true",
+										Reason: clusterv1.FieldValueRequired,
+									},
+									{
+										Rule:   "true",
+										Reason: clusterv1.FieldValueInvalid,
+									},
+									{
+										Rule:   "true",
+										Reason: clusterv1.FieldValueDuplicate,
+									},
+									{
+										Rule:   "true",
+										Reason: clusterv1.FieldValueForbidden,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "x-kubernetes-validations should have valid fieldPath for array",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "object",
+						XValidations: []clusterv1.ValidationRule{
+							{
+								Rule:      "true",
+								FieldPath: ".foo['b.c']['c\\a']",
+							},
+							{
+								Rule:      "true",
+								FieldPath: "['a.c']",
+							},
+							{
+								Rule:      "true",
+								FieldPath: ".a.c",
+							},
+							{
+								Rule:      "true",
+								FieldPath: ".list[0]",
+							},
+							{
+								Rule:      "true",
+								FieldPath: "   ",
+							},
+							{
+								Rule:      "true",
+								FieldPath: ".",
+							},
+							{
+								Rule:      "true",
+								FieldPath: "..",
+							},
+						},
+						Properties: map[string]clusterv1.JSONSchemaProps{
+							"a.c": {
+								Type: "number",
+							},
+							"foo": {
+								Type: "object",
+								Properties: map[string]clusterv1.JSONSchemaProps{
+									"b.c": {
+										Type: "object",
+										Properties: map[string]clusterv1.JSONSchemaProps{
+											"c\a": {
+												Type: "number",
+											},
+										},
+									},
+								},
+							},
+							"list": {
+								Type: "array",
+								Items: &clusterv1.JSONSchemaProps{
+									Type: "object",
+									Properties: map[string]clusterv1.JSONSchemaProps{
+										"a": {
+											Type: "number",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "x-kubernetes-validations have invalid fieldPath",
+			clusterClassVariable: &clusterv1.ClusterClassVariable{
+				Name: "var",
+				Schema: clusterv1.VariableSchema{
+					OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+						Type: "object",
+						XValidations: []clusterv1.ValidationRule{
+							{
```

Comment:
If we have more than one error case. Can we split them in separate test cases? Otherwise we don't really test which ones fail

Reply:
Added checks for all returned errors so this makes it easier to check them all in one test case.

Reply:
Perfect!